p2p/switch.go

```diff
@@ -381,6 +383,11 @@ func (sw *Switch) DialPeersAsync(addrBook AddrBook, peers []string, persistent b
 // DialPeerWithAddress dials the given peer and runs sw.addPeer if it connects and authenticates successfully.
 // If `persistent == true`, the switch will always try to reconnect to this peer if the connection ever fails.
 func (sw *Switch) DialPeerWithAddress(addr *NetAddress, persistent bool) error {
+	// do not dial ourselves
+	if sw.addrBook.OurAddress(addr) {
```

Comment:
Nice!

Reply:
Not sure if this is the right spot.

The thing we want to avoid is our address getting into the AddressBook. If it does, then we'll always be trying to dial ourselves, and we'll see logs marking this failure.

If we can ensure we don't get into the addrbook, or that if we do we are soon removed, then I think this doesn't need to be here